src/service/performance-impl.js

```diff
@@ -233,6 +269,48 @@ export class Performance {
     observer.observe({entryTypes: entryTypesToObserve});
   }
 
+  /**
+   * Bind to various page lifecycle events to capture visibility state changes.
+   * @see https://developers.google.com/web/updates/2018/07/page-lifecycle-api
+   */
+  registerLayoutJankHandler_() {
+    // The visibility state of the document.
+    let visibilityState = getVisibilityState(this.win.document);
+    // Whether the `lj` metric was sent.
+    let sentFirstJankScore = false;
+    // Whether the `lj-2` metric was sent.
+    let sentSecondJankScore = false;
+
+    const handleVisibilityStateChange = nextState => {
+      const previousState = visibilityState;
+      if (previousState === nextState) {
+        return;
+      }
+      visibilityState = nextState;
+
+      if (nextState !== 'hidden') {
+        return;
+      }
+
+      if (!sentFirstJankScore) {
+        this.tickDelta('lj', this.aggregateJankScore_);
+        this.flush();
+        sentFirstJankScore = true;
+        return;
+      }
+      if (!sentSecondJankScore) {
+        this.tickDelta('lj-2', this.aggregateJankScore_);
+        this.flush();
+        sentSecondJankScore = true;
+        return;
+      }
+    };
+
+    ['pageshow', 'focus', 'blur', 'visibilitychange', 'resume'].forEach(eventName => {
+      this.win.addEventListener(eventName, () => handleVisibilityStateChange(getVisibilityState(this.win.document)), {capture: true});
```

Comment:
I'd store a reference to the listeners added so you can remove them later.

Reply:
Done.

Reply:
It looks like you're adding a bound event listener and then removing the non-bound version.

Reply:
Good catch - I moved the binding to the constructor to fix.